osutil/stat.go

```diff
@@ -133,3 +134,33 @@ func RegularFileExists(fn string) (exists, isReg bool, err error) {
 	}
 	return true, fileStat.Mode().IsRegular(), nil
 }
+
+// ComparePathsByDeviceInode compares the devices and inodes of the given paths, following symlinks.
+func ComparePathsByDeviceInode(a, b string) (match bool, err error) {
+	info, err := os.Stat(a)
+	if err != nil {
+		return false, err
+	}
+	sysInfo, ok := info.Sys().(*syscall.Stat_t)
+	if !ok {
+		return false, fmt.Errorf("cannot get device and inode info for %q", a)
+	}
+	dev_a := sysInfo.Dev
+	inode_a := sysInfo.Ino
+
+	info, err = os.Stat(b)
+	if err != nil {
+		return false, err
+	}
+	sysInfo, ok = info.Sys().(*syscall.Stat_t)
+	if !ok {
+		return false, fmt.Errorf("cannot get device and inode info for %q", b)
+	}
+	dev_b := sysInfo.Dev
+	inode_b := sysInfo.Ino
```

Comment:
Avoid redefining error messages this way
```suggestion
        var devAndInode = func(p string) (dev, inode uint64, err error) {
	    info, err := os.Stat(a)
	    if err != nil {
	        return 0, 0, err
	    }
	    sysInfo, ok := info.Sys().(*syscall.Stat_t)
	    if !ok {
	        return 0, 0, fmt.Errorf("cannot get device and inode info for %q", a)
	    }
	    return sysInfo.Dev, sysInfo.Ino, nil
        }
        
	dev_a, inode_a, err := devAndInode(a)
	if err != nil {
	    return err
	}
	
	dev_b, inode_b, err := devAndInode(b)
	if err != nil {
	    return err
	}
```